### ISSUE TITLE
Node: SQL indexes are inexplicably missing

### ISSUE THREAD

**Issue by jcnelson** [blockstack/stacks-blockchain#2417]
The recent release sped up `GetBlocksInv` handling and block-download searching by creating some additional indexes on the `staging_blocks` and `staging_microblocks` databases.  However, the indexes are _not_ getting instantiated when the node boots up.

Shell trace:

```
$ sqlite3 /data/stacks/mainnet/chainstate/chainstate/chain-01000000-mainnet/vm/index 'PRAGMA index_list("staging_blocks")'
0|sqlite_autoindex_staging_blocks_1|1
$ sqlite3 /data/stacks/mainnet/chainstate/chainstate/chain-01000000-mainnet/vm/index "CREATE INDEX index_block_hashes ON staging_blocks(index_block_hash);"
$ sqlite3 /data/stacks/mainnet/chainstate/chainstate/chain-01000000-mainnet/vm/index 'PRAGMA index_list("staging_blocks")'
0|index_block_hashes|0
1|sqlite_autoindex_staging_blocks_1|1
$ sqlite3 /data/stacks/mainnet/chainstate/chainstate/chain-01000000-mainnet/vm/index 'PRAGMA index_list("staging_microblocks")'
0|sqlite_autoindex_staging_microblocks_1|1
$ sqlite3 /data/stacks/mainnet/chainstate/chainstate/chain-01000000-mainnet/vm/index "CREATE INDEX staging_microblocks_index_hash ON staging_microblocks(index_block_hash);"
$ sqlite3 /data/stacks/mainnet/chainstate/chainstate/chain-01000000-mainnet/vm/index 'PRAGMA index_list("staging_microblocks")'
0|staging_microblocks_index_hash|0
1|sqlite_autoindex_staging_microblocks_1|1
$
```

I noticed from the node logs that when I executed these commands, the time to process `GetBlocksInv` queries suddenly fell from about 2000ms to about 400ms (this is during the node's initial boot-up, so these times likely both skew slow [EDIT: it's now down to about 50ms at steady-state]).

**Comment by jcnelson** [comment]
I think I found out why -- there's a typo. The code has:

```
CREATE INDEX processed_stacks_blocks ON staging_blocks(processed,anchored_blcok_hash,consensus_hash);
```

It should have

```
CREATE INDEX processed_stacks_blocks ON staging_blocks(processed,anchored_block_hash,consensus_hash);
```

**Comment by jcnelson** [comment]
Looks like we also have a typo in the `snapshots` SQL for the sortition DB -- we're missing a `;`.

**Comment by jcnelson** [comment]
Update -- per @kantai, this is also due to us using `execute()` instead of `execute_batch()`.  The tables get created, but the indexes do not since they are separate statements.  We didn't notice this before because each `CREATE TABLE` statement is in its own string to be fed to sqlite, but that same string contains all of the subsequent `CREATE INDEX` statements for that table.

**Comment by lgalabru** [comment]
Yeah we also had an issue with 

```sql
CREATE UNIQUE INDEX index_block_hash_to_primary_key(index_block_hash,consensus_hash,block_hash);
```

fixed in #2420 with

```sql
CREATE UNIQUE INDEX index_block_hash_to_primary_key ON block_headers(index_block_hash,consensus_hash,block_hash);
```

**Comment by lgalabru** [comment]
Addressed with #2420
